tests/test_grimgrains.py

```diff
@@ -0,0 +1,81 @@
ike it') is a Japanese dish, similar to the American omelette, but the main difference is the variation of ingredients. Typical okonomiyaki are made with eggs, and often include meat or fish. Making it without meat is simple enough, but without eggs? Back when we lived in Tokyo, our experience in cooking with plants was limited, but now we've been doing it long enough that we can think of alternatives with ease.\nThe key ingredient? Chickpea flour. We make chickpea pancakes, and scrambled chickpea flour (resembles scrambled eggs) at home all the time. Chickpea flour is a staple on Pino, and works very well for okonomiyaki.\nNagaimo: If you're in a place were nagaimo (or yamaimo) is available, we highly reccommend adding it to the dish. It makes a fluffier pancake. Although we've made okonomiyaki without nagaimo before, so if you can't find it know that it will work and be very delicious anyway. It imparts little flavor, all it does is add nutrition and texture. Nagaimo, unlike most potatoes, can be eaten raw. However, it is best to handle the nagaimo with gloves, or to soak the peeled tuber in a vinegar-water solution to neutralize irritant oxalate crystals found on their skin. Nagaimo are low-calorie, high in protein, and have potassium, zinc, vitamin C and more. The texture of grated nagaimo can be off-putting, it looks like a regular tuber when whole, but when grated it becomes slime, almost liquid. This sort of texture is well-liked in Japan and referred to as being 'neba neba' (slimy). This texture present in many other foods like okra and nattou. This texture makes it an ideal egg alternative, it can be used to make deserts when baking.\nAonori: Aonori is another obscure ingredient, but again, it can be omitted, although it tastes really amazing with it. We made okonomiyaki without it when we were in Majuro, because it simply wasn't available, so we used finely cut nori instead. Obviously, this isn't a perfect substitution, because aonori is sweet and tastes nothing like nori. However, nori is still very delicious and pairs well enough with the okonomiyaki.\nOkonomi sauce\nIn this recipe, we don't use true 'okonomi sauce'. Why? Because we don't use many pre-made sauces, we prefer to make my own. Okonomi sauce requires many ingredients, and honestly, the sauce we've made works really well in this recipe and makes a good okonomi sauce alternative.\nIf you want to make your own, you can mix 7g (1 1/2 tsp) sugar, 45g (3 tbsp) ketchup and 45g (3 tbsp) of vegan worcestershire sauce. If you are like us, and don't care to buy pre-made sauces but want to avoid buying both ketchup and worcestershire sauce, you can make these too:\nWorcestershire sauce: combine apple cider vinegar, water, soy sauce, sugar, mustard powder, onion powder, garlic powder, cinnamon and black pepper in pan, bring to a boil and cook for a minute, then let cool.\nKetchup: Using some fresh tomato sauce may be enough, otherwise add a bit of sugar and apple cider vinegar to it.",
```

Comment:
I think this `description` output is a bit too long; that field should generally be a brief summary of what the recipe is (a bit like an extended `title`).

It does contain a nice description of what Okonomiyaki is in this case, but I think it'd be OK to remove the `description` method entirely if we can't find alternative text to use on the page (I couldn't see any in the HTML - the `meta` description field is too generic).

Reply:
(I should have noticed that during my first review.  my mistake)

Reply:
In other recipes it's a lot shorter, I picked a long one to better test getting the whole text, but maybe I could limit it to the first paragraph?

Reply:
Mmm.. I'm not sure.  Those paragraphs seem like relatively unstructured article text, so I don't think they're guaranteed to provide a description of the recipe.

Reply:
(a short/overview description, I mean - what the `description` field is meant to represent, I think)

Reply:
I'm not sure this'll help, but to try to get a better idea of what `description` is about:

* It originally -- and usually -- represents the [`schema.org` description field](https://www.schema.org/description) -- but that isn't well-defined.
* The `schema.org` definition mentions [Dublin Core's `description` field](https://www.dublincore.org/specifications/dublin-core/dcmi-terms/#http://purl.org/dc/elements/1.1/description) as an equivalent attribute.
* Dublin Core mentions: "Description may include but is not limited to: an abstract, a table of contents, a graphical representation, or a free-text account of the resource."

In our case "the resource" is the recipe contents (title, ingredients, instructions, some other key fields), so I think `description` should be an overview/summary of that -- but not content from within the recipe itself.  Maaaybe the first paragraph usually approximates some kind of summary, but I think it's ambiguous-enough that we shouldn't risk it.

(that's a longer-winded attempt to explain roughly what I had in mind with the previous comment)

Reply:
Thanks for the explanation! I understand what you mean and I agree

Reply:
No problem - thank you!